### PR TITLE
CLN: remove deprecated device references

### PR DESCRIPTION
**Pull request by klauer**
## Description

- Removed deprecated ``happi.containers.Device`` container.
- Removed deprecated methods:
  - ``happi.Client.create_device``
  - ``happi.Client.add_device``
  - ``happi.Client.find_device``
  - ``happi.Client.all_devices``
  - ``happi.Client.remove_device``
  - ``happi.SearchResult.device``

## Motivation and Context
* v1.14.0 is tagged
* v2.0.0 is on the radar

## How Has This Been Tested?
Test suite

## Where Has This Been Documented?
Issues + release notes

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [x] Test suite passes on continuous integration (Travis CI)
- [x] Ran ``docs/pre-release-notes.sh`` and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Review by JJL772 — APPROVED** [review]
looks good to me, aside from one very minor nitpick

File: happi/utils.py [hunk 1]
```diff
@@ -93,34 +90,6 @@ def inner(value):
     return inner
 
 
-def deprecated(message: str):
```
[the remark this answers is not in the document]
Comment (ZLLentz) [reply happi/utils.py:96]: This is the only thing I noticed too

**Review by ZLLentz — APPROVED** [review]

File: happi/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (klauer) [reply happi/utils.py]: OK, adding it back in for future use: 9a2c6bb

[the remark this answers is not in the document]
Comment (JJL772) [reply happi/utils.py]: I guess you could say this function was deprecated :trollface:

**Review by tangkong — APPROVED** [review]
LGTM as well 👍 I'm excited for the chaos this will cause

File: happi/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ZLLentz) [reply happi/utils.py]: I mean, we could deprecate the deprecator- but it isn't harming anyone the same way confusing names do

[the remark this answers is not in the document]
Comment (klauer) [reply happi/utils.py]: Should I make another deprecator helper function to deprecate the deprecator?

[the remark this answers is not in the document]
Comment (ZLLentz) [reply happi/utils.py]: You could make the deprecator wrap itself after it has been defined- can't use the decorator syntax of course.

[the remark this answers is not in the document]
Comment (JJL772) [reply happi/utils.py]: I think what we need is a non-deprecated deprecation decorator to deprecate old deprecation decorators

**Comment by klauer** [comment]
🦗

**Pull request merged**